components/common/main-content/MainContent.js

```diff
@@ -16,29 +16,62 @@
  * under the License.
  */
 
-import * as React from 'react';
-import ReactMarkdown from 'react-markdown';
-import remarkGfm from 'remark-gfm';
-import rehypeRaw from 'rehype-raw';
-import remarkExternalLinks from 'remark-external-links'
+import * as React from "react";
+import ReactMarkdown from "react-markdown";
+import remarkGfm from "remark-gfm";
+import rehypeRaw from "rehype-raw";
+import remarkExternalLinks from "remark-external-links";
+import { getHighlighter } from "shiki";
 
 String.prototype.hashCode = function () {
   var hash = 0,
-    i, chr;
+    i,
+    chr;
   if (this.length === 0) return hash;
   for (i = 0; i < this.length; i++) {
     chr = this.charCodeAt(i);
-    hash = ((hash << 5) - hash) + chr;
+    hash = (hash << 5) - hash + chr;
     hash |= 0;
   }
   return hash;
-}
+};
 
 export default function MainContent(props) {
-
   const content = props.content;
   const codes = props.codes ? new Map(JSON.parse(props.codes)) : new Map();
 
+  // Synatax highlighting
+  const HighlightSyntax = (code, language) => {
```

Comment:
We do not need to do any changes in this file. Also the changes you have done creates rendering issues in some pages which uses this component.

Can you revert the changes of this file?

Reply:
By removing this, syntax highlighting will be disabled in some pages. By using conditions, it can be fixed.
Can you please try updated code in new PR.

Reply:
Looks like you were right. We need that function to add syntax highlighting in pages where the `html`s are used as below. I am sorry I missed that scenario.
![image](https://github.com/ballerina-platform/ballerina-dev-website/assets/7709843/b50791bc-7158-408f-a3af-01a97f9b12c6)

But your previous fix (with changes to mainContent.js) gave an error in this page https://ballerina.io/learn/generate-code-documentation/
There is a nested code block which is what causing the issue I think.
Can you check that?

Reply:
I observed that only `ballerina` language is used in this function. All other language code comes from `props.codes`.
I guess I can remove all other languages from this function.
If I keep other languages too then it is giving some error in other file.
![HighlightSyntax function](https://github.com/ballerina-platform/ballerina-dev-website/assets/99467494/35f09dd4-ba56-4a5b-a21f-6c5fc6dbe339)

Please visit the new PR on this issue.

Reply:
Getting an error. 
![image](https://github.com/ballerina-platform/ballerina-dev-website/assets/7709843/503f6f4f-1207-4bd8-9049-098ce923a978)

Reply:
It was working fine in my device. I added try cache block for any possible error due to this function.
Please try new PR once.

Reply:
@JayPokale I think the main issue is we have some old release notes in `html` format and I don't think we can re-use the `mainComponent.js` as expected. I think we have to go with your original fix (https://github.com/ballerina-platform/ballerina-dev-website/pull/8067/files). I tested that again. But noticed that the headers & the toc were missing in that fix.
![image](https://github.com/ballerina-platform/ballerina-dev-website/assets/7709843/b179fe78-87c7-483e-b8a0-ecc196611c35)

Reply:
New code works fine due to try cache block.
I am also trying another method for the same issue.
I also observed one issue. If you are offline then HighlightSyntax giving fetching some error. I fixed this too in new PR.
Please check my new PR.

Reply:
Getting this error now. 
![image](https://github.com/ballerina-platform/ballerina-dev-website/assets/7709843/aeb90728-972e-4359-b96c-6cc05c7172bb)

Did you check your old fix (https://github.com/ballerina-platform/ballerina-dev-website/pull/8067/files) as well?

That nested code block related issue will be tracked via https://github.com/ballerina-platform/ballerina-dev-website/issues/6656

Reply:
Now I fixed it by old method

Reply:
Please test my new PR. I used old fix in this.